lib/publiccloud/provider.pm

```diff
@@ -378,19 +378,19 @@ C<proceed_on_failure>  Same as timeout.
 sub create_instances {
     my ($self, %args) = @_;
     $args{check_connectivity} //= 1;
-    $args{check_guestregister} //= 1;
+    # $args{check_guestregister} //= 1;
     $args{upload_boot_diagnostics} //= 1;
     my @vms = $self->terraform_apply(%args);
     my $url = get_var('PUBLIC_CLOUD_PERF_DB_URI', 'http://larry.qe.suse.de:8086');
 
     foreach my $instance (@vms) {
         record_info("INSTANCE", $instance->{instance_id});
-        if ($args{check_connectivity}) {
-            $instance->wait_for_ssh(timeout => $args{timeout},
-                proceed_on_failure => $args{proceed_on_failure}, scan_ssh_host_key => 1);
-        }
-        # check guestregister conditional, default yes:
-        $instance->wait_for_guestregister() if ($args{check_guestregister});
+        # A next VM-up error causes test to stop
+        $instance->wait_for_ssh(timeout => $args{timeout}, proceed_on_failure => $args{proceed_on_failure}, scan_ssh_host_key => 1)
+          if ($args{check_connectivity});
```

Comment:
there is no actual change , you just rewrite it from 
if (condition) {
    function;
}

to 
function if(condition);

I don't care in which way it will be written but I find this confusing for review and also later when someone will check the diff. So if you really want this change to stay please do separate commit for it or simply revert

Reply:
I even don't see much the sense to run conditionally this important check that VM is up, that should always be done, but ok I restore the structure.